### DISCUSSION TITLE
Log in with username instead of email

### DISCUSSION THREAD

**Discussion by brunoribei-ro** [directus/directus#4958] [Archive]
I'm working on a project that requires a user to log in with his **CPF** _(which is the Brazilian social security number)_
Despite knowing how to add this new field to the user collection, I would like to know **if it is possible to make use of it instead of email when calling auth/login endpoint.**

 Does anyone has already done something similar to it? 

**Thank you all!**

**Comment by MonksterFX** [comment]
It is hardcoded here for email: api/src/services/authentication.ts
So you can theoretical change it in Code.

**Comment by rijkvanzanten** [comment]
@brunoribei-ro You can't override the auth to use a non-email field for login. That being said, you should be able to create a custom endpoint that's basically a copy of `/auth/login` except it checks for a custom field instead of email

**Reply by alcaprar** [reply]
This seems similar to this suggestion: https://github.com/directus/directus/discussions/6670#discussioncomment-1060749
